### ISSUE TITLE
Request: Update shows before search

### ISSUE THREAD

**Issue by boredazfcuk** [SickGear/SickGear#1362]
This is a feature request rather than an issue.

There have been a couple of shows recently which don't have episode metadata until the show airs. This means all the episodes are named `TBA`. SG has struggled to find these episodes.

Would it be possible to get an option to update the episode's metadata immediately before searching for it?

**Comment by JackDandy** [comment]
> SG has struggled to find these episodes.

Not due to an episode name as SG does not use episode name to search given the fragility of that idea *shudders*.

A force update in `/view-show` or `/manage/bulk change` will update the shows meta whenever run. If you want to automate that, you can trigger it via the SG API. It will affect performance though, and has no bearing on any SG processing, which is why SG is designed to do that once a day per your config. Of course, this means that a show can be processed, have meta set one way, and then not be updated from those values, until whenever the next change propagates into the TVinfo source API (which is not necessarily the same as TVinfo website).

**Comment by boredazfcuk** [comment]
> Not due to an episode name as SG does not use episode name to search given the fragility of that idea _shudders_.

Yeah, I wasn't sure if it was the case, but thought it would be worth mentioning.

> A force update in `/view-show` or `/manage/bulk change` will update the shows meta whenever run. If you want to automate that, you can trigger it via the SG API. It will affect performance though, and has no bearing on any SG processing, which is why SG is designed to do that once a day per your config. Of course, this means that a show can be processed, have meta set one way, and then not be updated from those values, until whenever the next change propagates into the TVinfo source API (which is not necessarily the same as TVinfo website).

This is why I was wondering if it could update the information for the single episode. It ensures that the file/nfo is correctly named in the library. Currently this isn't the case.

**Comment by JackDandy** [comment]
Okay, let me try again,

> It ensures that the file/nfo is correctly named in the library. Currently this isn't the case.

But it _is_ correctly named. It just doesn't reflect the name as you would like it to at the exact time in space that you know the name happens to no longer be "TBA" - and that distinction is crucial to why what you suggest sounds good on paper, but is not technologically practical. 

When a new episode drops, there is nothing to say that the TVinfo source has **proper** data at the time which you are suggesting to add logic into the SG auto process. Indeed, often, this is simply _not_ the case.  Worse, the API does not expose user updated data for quite some hours after you suggest it could be available for use.

**Comment by boredazfcuk** [comment]
The four shows that have had this problem (so far) are incredibly popular. I would have expected them to have been updated within minutes of the information being made public, Certainly before the shows have ended. Due to time zone differences, I can't say for sure though, as I'm not available to check the metadata source, but the data has always been correct when I have manually checked it around three or four hours after air-time.

The only problem I can see is if, like you say, there is a delay between the website being updated and the API reflecting these changes. I will look into triggering a show update via the API like you suggest though.

**Comment by boredazfcuk** [comment]
I've now configured a script to update all the three shows I've had the problem with and another four I expect to exhibit the same behaviour.

I've scheduled it to run around 5 minutes after each show finishes airing. Hopefully this should fix the issue.

**Comment by JackDandy** [comment]
Hiya, I have circled back to this thread and have an idea I would like to try, sooo.... what are the names of the popular shows whose ep names are consistently TBA?

**Comment by boredazfcuk** [comment]
Hi,

Sorry for the slow response. I started a new job in November so I've been working flat out at that. Kinda ignoring everything else in life while I get up to speed.

It was affecting mainly the Disney/Marvel stuff from what I remember. I've not noticed in a while though as I've stopped watching most stuff. Started the Netflix-Marvel shows on Netflix in the summer and still haven't finished watching all those yet.

Thanks.
